tsdb/engine/tsm1/file_store.go

```diff
@@ -527,8 +527,8 @@ func (f *FileStore) BlockCount(path string, idx int) int {
 // locations returns the files and index blocks for a key and time.  ascending indicates
 // whether the key will be scan in ascending time order or descenging time order.
 // This function assumes the read-lock has been taken.
-func (f *FileStore) locations(key string, t int64, ascending bool) []location {
-	var locations []location
+func (f *FileStore) locations(key string, t int64, ascending bool) []*location {
+	var locations []*location
```

Comment:
The locations are actually modified as we seek.  When it switched from a pointer to non-pointer, it ended up causing blocks to get decoded more than they were supposed to, but it was not noticeable because we always `Deduplicate` them so the result was always correct either way.  When it was not a pointer, marking the block (or sections) of the block that had been read got lost in-between `Read*` calls.

I had to switch it to a pointer again because `Deduplicate` is no longer called and it ends up returning the points in a block multiple times.